test/app.spec.js

```diff
@@ -70,193 +72,371 @@ describe('JHipster generator kafka', () => {
                     })
                     .withPrompts({
-                        generationType: constants.BIGBANG_MODE,
-                        components: [constants.CONSUMER_COMPONENT, constants.PRODUCER_COMPONENT],
-                        entities: [FOO_ENTITY],
-                        topic: constants.CUSTOM_TOPIC,
-                        topicName: CUSTOM_TOPIC_NAME
+                        currentEntity: constants.NO_ENTITY,
+                        currentPrefix: COMPONENT_PREFIX,
+                        currentEntityComponents: [constants.CONSUMER_COMPONENT, constants.PRODUCER_COMPONENT],
+                        continueAddingEntitiesComponents: false
                     })
                     .on('end', done);
             });
 
-            it('should put a custom topic name in application.yml', () => {
-                const { applicationYml, testApplicationYml } = loadApplicationYaml();
-                assertTopicName(applicationYml, testApplicationYml, FOO_ENTITY, constants.CUSTOM_TOPIC, CUSTOM_TOPIC_NAME);
-            });
+            itGeneratesBasicConfigurationWithConsumerProducerWithAnEntity(COMPONENT_PREFIX);
+            itShouldUpdatesPropertiesWithDefaultValue();
+            itShouldTypeClassesWithClass(COMPONENT_PREFIX, 'String', COMPONENTS_CHOSEN.all);
         });
-    });
 
-    describe('with the incremental mode', () => {
-        describe('with only a producer for a single entity', () => {
+        describe('with a given offset and polling timeout', () => {
             before(done => {
                 helpers
                     .run(path.join(__dirname, '../generators/app'))
                     .inTmpDir(dir => {
                         fse.copySync(path.join(__dirname, '../test/templates/message-broker-with-entities-1st-call'), dir);
                     })
                     .withPrompts({
-                        generationType: constants.INCREMENTAL_MODE,
-                        currentEntity: FOO_ENTITY,
-                        currentEntityComponents: [constants.PRODUCER_COMPONENT],
+                        currentEntity: constants.NO_ENTITY,
+                        currentPrefix: COMPONENT_PREFIX,
+                        currentEntityComponents: [constants.CONSUMER_COMPONENT, constants.PRODUCER_COMPONENT],
+                        pollingTimeout: 500,
+                        autoOffsetResetPolicy: constants.LATEST_OFFSET,
                         continueAddingEntitiesComponents: false
                     })
                     .on('end', done);
             });
 
-            it('should generate default, producer, serializer, deserializer, serde and resource', () => {
+            itGeneratesBasicConfigurationWithConsumerProducerWithAnEntity(COMPONENT_PREFIX);
+
+            itShouldUpdatesPropertiesWithGivenValue();
+        });
+    });
+
+    describe('with an existing previous generation', () => {
+        describe('with --force --skip-prompts options', () => {
```

Comment:
```suggestion
        describe('with --skip-prompts options', () => {
```

Because Yeoman use the force option by default :)

You can remove it in the describe of other tests with `--force` in it.